### ISSUE TITLE
Idempotency problem when using a list with interpolated strings

### ISSUE THREAD

**Issue by abelbraaksma** [fsprojects/fantomas#2242]
Issue created from [fantomas-online](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKEIBtEBcwDMwALeAJwGlEBPMAXgB1T6A7DNsUxAQ3gHlnMVACIBLaDgDaAEnqRgJCtQC%2BKAAyyANGFkANALQBZRAGdjXAOaIAkkNmMW7R04wy5CylRUBGTdsgBBAFccIlhSEQAvLhwRWGZZAF17ZkgNcChjXBjmc2NUMAl0iFBHSCkcKgAHZDRIEWYcVKKMMoA3LkxAmoLmtlU050grZiQGgGVI5AGnABZesATmpWni%2BbKK6vy6hqbSqXbO7sLnLxX2SAMuAA8AGXrEG8QckN2nLwAmVXnFtmXmkvO5Sq3UgTz4%2BDuzEQYwq2FebDaHS6%2BWOTneZwRAFERrwIfd4edMPhIN8lmcAQigZtagAjWCwTAEiCIw4o%2BYQADMGJaw0ypBwADF6h0AHKIADumHx3IgOFIXVJv3JaypILpDKZUH2SKO7LAMxlUDGiAAtmIGXF-DhseDIVM9fgOplFRg-mwKS1VVt1YzuSzkWhUY4AKyGyBjSpcaCIABCiHwYUQAAUuKQuCbcIhSJrZfLEC6IG6MB7mV7afTfSqDgGes4AGxhiNR2PxxM3WDirPQLiZYatWDdmJxHNgOUKxw-V3KvYbNUVzX%2B3XOADsjcj0bjCc4AFVKtVSN3e8x%2B4PYilDY7MM6J2T-irZ97537tazA3qABxr5ubxMAYUwPbGL%2BcTGGO4hhCOl7Xuwk6FtOgIPuWGrPtWS5OAAnF%2BG6tpwRgmjSWaQU6%2BY3kqd4zsCj7IVWOpsic-R6uG64tluiDAZgw4XsRBZgEWqwUdSUA%2BguL41kG7BeKcjFNtG-j4DgWbASaJpcCOY4kTBt7uvelFIZWeyoXRbzotJzE-pwxpmnAHHng63GkVO5EIbpQlPjRr61m8XKmc2ckKaQlnmjZal5jxfFgCWWqIa51EGbRb4nAajHDKMOD8AAKqQVAAOoiC8XFXhpbCwbx8GUtFkDCSh8WeY4Xihj5smkLAgQjEIiBSma-kheOmlkdpAkgvUjTVR54lsF4DaMZcVxWPg6VEE8mKFWMoT8rl8D5XqMxfA5cFOeVLnbCN7liXqXirtN1zDPgIhXLw%2B7RGEmJXJUnCmGeI7BrtfWOQNzmCcdImGQlbyfldVwAEqIHApDwBtW3ODtYVlZ6FXDcDNXjRgXiYRD0Ow-AIqBPhWa4lYCkmnkhpeCjB1o0dJqBJgMRSlCAphCpOD%2BellGjWdzifGGBNhPABjM6z9wc6QXPdWG0BEKmUb%2BQA%2BuKIibUQJJ7aV9OlujOz82hjjvFJgwzf4pBplQvCkHcoEI1rhrIzr4WResR0Y0bRkmyZ5vXJb1u2-bODE6TpDk5T1PnXT-2HYDTMsyIbOINLstZrzmze6DJveYMgdcDbdsiKB4tJynafRHLjEK0r4hZmrGsvLHxY6YDXuncb7DvEl-tXAAajqMb1PA9TmI7I7vj9xVaa3g1bB3cVjXq7wNX3AqteIZ7DyMY8T8708YCVbtt0Nhudz73dTX3QiwDgADiuAvW9JjGGe%2B96t9Lf8QDc6xYCINard0uubCWyd7gxg4tAAA1jGNMMDcDGH4GMdMbEGQk1ss4KCRUj6zx-vHP%2B%2BkAFYxXuDQYYpJQQNwJ2J4mdEDtVuswPKZ5-AjDwgRUg0csH2V%2BvtOODNAZVQvjnbueNBiUEQJUOaC0njDBQRmO0RFCrfwiqfBe58l4CycByBifdlpmmMEQD%2BSND4QGPqjfWR0hGaK7mwDkZsnDhjHtgS25gSZPBwNlRANIDAIHtNw5RrsLFRSsW5Gxl87F%2B0cf4KU5hmAb2YFvOIEw4nRECJwdKsAUoeOiJ9Aq0EZ79Tnr-KiRDKSAOxpyPO0SWZZmYLk1oDw4jmHYVmBh9RmEgSUQU3BRT8ECMIZjZezgOS90cWXGIdobjphpPALg-5YBvxyBQlO3ScFmLwao%2BeekhlaMcByNejjRBmBpNgfRJciBjCoA0a4ayVHuzLDFMpnoKl6g5NfRxEipEjA8cMOBXBElOzsoE3hut%2BGWMEWE4hwztEgMcZAgF0C7TGEti1EYYpQKIDFmAlOz93pvy6YadS9y1E7OzkAuxZD4WpnMvQku0BwhmnqQpeA24mFxHatAACaYhyYKcNgkl2ynm7NsRgDkYjHGQwsPU5q0C7lBL1iEyF-9ykkJMY2OUYgcC%2BKQPK0FSh0gJGmHUYwAo375AFUoIAA)

#### Before formatting, code was compiling:

```fsharp
let f hdrKey =
    readOnlyDict[$"{hdrKey}:0", "X-MessageID"
                 $"{hdrKey}:1", "Authorization"]
```

#### Formatted code

```fsharp
let f hdrKey =
    readOnlyDict[$"{hdrKey}:0", "X-MessageID"
    $"{hdrKey}:1", "Authorization"]

```
            

#### Reformatted code (note the missing semicolon!)

```fsharp
let f hdrKey =
    readOnlyDict[$"{hdrKey}:0", "X-MessageID" $"{hdrKey}:1", "Authorization"]

```
            
#### Problem description

Fantomas was not able to produce the same code after reformatting the result.

#### Extra information

- [x] The formatted result breaks my code.
- [x] The formatted result gives compiler ~warnings~ **errors**.
- [x] I or my company would be willing to help fix this.

#### Options

Fantomas master branch at 2022-05-09T12:59:04Z - f2e3c0898acc865e05bb3d94510daadf797a234c

Default Fantomas configuration

* I've seen this fail with Fantomas 4.7.0
* Exactly the same code with VS 2022 Extension uses Fantomas 4.6.0 and succeeds and leaves the code as-is
* In other words, **seems to be a regression**.

<sub>Did you know that you can ignore files when formatting from fantomas-tool or the FAKE targets by using a [.fantomasignore file](https://github.com/fsprojects/fantomas/blob/master/docs/Documentation.md#ignore-files-fantomasignore)?</sub> <kbd>yes ;)</kbd>.

**Comment by abelbraaksma** [comment]
#### [Alternative, failing code:](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKEIBtEBcwDMwAPMAXjACdEBDeAeQDtMBPAEQEtocBtAEgB1IwIgF8UABkEAaMIMEBuMAKGiUARmmzIggLr8K-BpCngoAZ1w52DAOZnUYbqYigMGSLxzMADsjSRrHGNnd14AN2pMAFc-RxC3cRM3N0gASQYkBhwAZXYAL2Qk5IgAFniwHRCRIpdyjy9fBwCs4OKPCOjYp2KINRrkyABZaiIAGWtEUcRbHAALVp61ACZxcsq3apDXAc8fWMhp%2BnxxhkRsr2wFlPDImIdunqX%2BlIBRDLpjiauBzHxINaq-W21wa%2BwARrBYJhvhB2rcuuUIABmZ7udIWCg4ABi1kiADlEAB3TBfVEQHAUGIAjZAuq7Rr%2BCFQmFQG6de6IsAlMlQbKIAC2nChsAYAEEcG8jidCpz8JELNSMJs3MDQqCmkzoai4ey0A9igBWHmQbLeajQRAAIUQ%2BFgVAACtQKNR%2BbhEBQWeTKYhFRBlRhVbD6eDIVq6R07nrOQA2Y2m81Wm12yawQnu6DUCzpMKwDNWEWesAUqnFdZK2ltYMa0MsnWRuI9MAAdjjZot1ttVAAqt5fBQM1mGDm8%2BwCzy5ZgFaXAVs6erGTXtWz6-rkgAOVsJjvJgDCmEzZh3IrMxa4dsLE6nyTLforO3nUE1teXCMbAE5N%2B2k1RBgKwe6L3lH1pxpWdKwfSAnyXCNX0WRJORNNtE07RAj0wMdZSA30wH9WpwL2atmWg%2BEOUbNQ%2BgQ%2BMLVFfAcHdI9%2BX5ahC2LYDrxnFU5wIhciPDEiozIp5KKQ7cqD5QU4HQoxxywkDyzA%2B9uMfRc%2BN1BtFhRYSExouiKHEoUpJY71sNwsBA1ZCCoNUldOTUbkEPSTIcEYAAVChmAAdXYOZAMnNi3BvHC7xBJTIJUysYNIxYjS06iKFgKIMlYRASUFXSjJLdjQM4-CGSgQJn0igTFljBDhiIVJ8Bc2ZphePzslmO0cC8%2BAfJ5EpVjk28FJCvLmiCYi1NXNw1BbMqRnSfB2CIOg%2B2oHA7ReIhvCoMwzFHaTOQNTqsvknLFL6grBpssiN3GogACVEDgCh4BatrOQ6kzgrVUKjus2DijUD9zqum74DxKJ%2BX-CgPlSOj%2BXsHk1GenrXr6-kokwKwSVOLE7SYnBdJcgjjs%2B5IVmNP67XgQYkZRiZ0YoTH0uNaBZidc1dIAfUJdhWvmWH9t6-Z3oi-j1OKJYKMbIYRlFChnWYOgKHGE97vmdqdoCjiAy4w6WjxqKhaE0XyolqWZblnBAeB90wYhqHbK5tXcv2RHkfYVHECpmn3RxxoteKoXNNFg3qGl2X2BPMnHed135tphD6cZrh3VZ9mfJtvCDt5zWPu1gn7L1kYADV4Utax4GsGwFcLNdlYwQLTPM%2Bo3vT-mhs5JYYpzogsUSrgNsLjIS7LpXk7M9W04GjPvYJ0q29YWAcAAcVwJaVsQNaNv7rbK4gauXqDSzwp2IrBYJsa9fJp2JktdDoAAa0tZ1r9wMxGGyF1UKhIHNsbS9-Kr1WU55wiwyNxOo8M6osCTEnPrgNM0wPaIGSlNBg3kNqigyL%2BM2FAraf1krtbq3N4Yhl4kA-Gbglg-VFgAaUQIgbwlVqrTHSM-V00pfJXhVtlW2qcmh833gLYaGAkTwTbnVQUZhZhr0bE9LqQU4Y71ClZIhmc3BIhFj0E0JdsASxsEDaYzVEBgkGAgGUWC-KD1rlWHigCeFN0bEiXWqjRQkhsAwDuDAu4ilyE4%2BaUQqAuVgI5HR80NosO-pvX%2BQ87YAMKrwzkSJfb2ORu6BggSwgplsGgkG8DrBIOPME0xw9Ile0Pko7OqjQ5WGlKMF0YJ4DUD3LAdathwHO1yVImu%2BSLFROsT0JErdVEcDMNQME2BhHB1mNkZgWQRgtJwdIvBsi%2BryKscA4oSJJ6qModQ-xWR0i32oK4xWmETGtO3hZORe9rgHz4ciY%2BqiL57KvtKMwEsEoZAJCeRApNT7O0XqtdaOSeSsTyREjphSrlgCRKA25TpRJwODtACg7BBRJLovALsiCRTJWgPuZ0%2BYP49C-kCzhIKx5FP4WQ1RF1qCePilfaZbC9ocP-sShR483AdTjBSTgOADFIDpT-DYpgdBFACGYLE60HAEpEEAA)

```fsharp
    let f x = readOnlyDict[$"{x}:0", ""; $"{x}:1", ""]
```

#### After formatting, breaks:

```fsharp
let f x =
    readOnlyDict[$"{x}:0", ""
    $"{x}:1", ""]
```

**Comment by nojaf** [comment]
Hello, thank you for raising this.
I left feedback on how to fix this in [#2158](https://github.com/fsprojects/fantomas/issues/2158#issuecomment-1078725550).
Are you interested in submitting a PR?

**Comment by nojaf** [comment]
> In other words, seems to be a regression.

That is debatable btw. In F# 6 the new indexer syntax gives recontextualizes what you wrote.

```fsharp
> readOnlyDict;;
val it:
  (seq<'a * 'b> -> Collections.Generic.IReadOnlyDictionary<'a,'b>)
    when 'a: equality
```

`readOnlyDict`is a function call, so it is weird that you are trying to index into it with the `$"{x}:0", ""; $"{x}:1", ""` expression. Fantomas does not know what your code is doing, so it saw an index expression and restored it as such.

The only bug I see here is how it is wrongly handling a multiline expression inside the index.

**Comment by abelbraaksma** [comment]
> That is debatable btw. In F# 6 the new indexer syntax recontextualizes what you wrote.

@nojaf Well, I mentioned that after the observation that Fantomas 4.6 tackles this code correctly somehow. The project is F# 6.0 indeed, but running it with commandline Fantomas 4.7 showed the problem.

> so it is weird that you are trying to index into it with the `$"{x}:0", ""; $"{x}:1", ""` expression.

This is an interesting observation. I didn't consider this. So I tried a few things in F# Interactive, and I get a compile error each time, esp. w.r.t. it being an indexed property.

But, in the "normal" `.fs` file, it gets compiled just fine. Here's what I simplified it to, which compiles fine in a .NET 6.0 project, but does NOT compile when send to Interactive:

```f#
let configureCorsPolicy x = Seq.iter ignore x
let hdrKey = "foo"

configureCorsPolicy
    readOnlyDict[$"{hdrKey}:0", "X-MessageID"; $"{hdrKey}:1", "Authorization"]
```

**Comment by abelbraaksma** [comment]
Sorry, I stand corrected. It wasn't an error in FSI, it was a warning:

> info FS3365: The syntax 'expr1[expr2]' is used for indexing. Consider adding a type annotation to enable indexing, or if calling a function add a space, e.g. 'expr1 [expr2]'.

**Comment by nojaf** [comment]
Thanks for following up on this.

**Comment by abelbraaksma** [comment]
The compiler is a bit weird here. I removed the `#nowarn`, upped the warning level to `5`, but it doesn't report on it, except for in the editor itself when hovering. This is odd, since I did get the warning with a slightly different syntax.

That's clearly F# compiler territory and not so much Fantomas territory.

I might have a look one of these days in submitting a PR to fix the underlying bug where the alignment gets screwed up in multiline statements.

**Comment by nojaf** [comment]
> The compiler is a bit weird here

Well, one other thing to note is that Fantomas is using a slightly newer version of the compiler bits.
In the past, we referenced the FCS from NuGet, so depending on your .NET SDK version, there might be a difference there.

For a few days now, we are using our own FCS version, which is tied closely to the `main` branch of `dotnet/fsharp`. (See https://github.com/fsprojects/fantomas/issues/2217).

All of this isn't really a problem unless there are some more significant changes like the new index syntax.

> I might have a look one of these days in submitting a PR to fix the underlying bug where the alignment gets screwed up in multiline statements.

Thank you, check out our [Contribution Guidelines](https://github.com/fsprojects/fantomas/blob/master/CONTRIBUTING.md#getting-started-with-fantomas) to get started.

**Comment by nojaf** [comment]
Fixed by https://github.com/fsprojects/fantomas/pull/2446
